### PR TITLE
Remove more Symfony 4 specific code

### PR DESCRIPTION
**Pull request by glaubinix**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | missed some places in https://github.com/php-http/HttplugBundle/pull/441
| Documentation   | if this is a new feature, link to pull request in https://github.com/php-http/documentation that adds relevant documentation
| License         | MIT

#### What's in this PR?

Remove more Symfony 4 specific code

#### Why?

Support has been removed https://github.com/php-http/HttplugBundle/pull/441

#### To Do

- [ ] Do you want to keep around the `httplug.client` service? Then I move it to the services.yaml.

### CODE REVIEW

File: src/DependencyInjection/HttplugExtension.php [hunk 1]
```diff
@@ -60,11 +59,7 @@ public function load(array $configs, ContainerBuilder $container): void
 
         $loader->load('services.xml');
         // TODO: Move this back into services.xml when we drop support for Symfony 4, or completely remove the service in the next major version.
-        if (Kernel::MAJOR_VERSION >= 5) {
-            $loader->load('services_legacy.xml');
-        } else {
-            $loader->load('services_legacy_sf4.xml');
-        }
+        $loader->load('services_legacy.xml');
```
Comment (dbu) [inline src/DependencyInjection/HttplugExtension.php]: we could completely remove the legacy services and mention in the changelog what the new names to use are. or prefer to do that in a separate MR?

Comment (glaubinix) [reply src/DependencyInjection/HttplugExtension.php]: Sure, I can do this here. I take it, this means to also remove [the service alias in the services.xml](https://github.com/php-http/HttplugBundle/blob/1b16d97ad8d572211f61135139b367bf47e7d7ec/src/Resources/config/services.xml#L50)?

Comment (dbu) [reply src/DependencyInjection/HttplugExtension.php]: `Psr\Http\Client\ClientInterface` should still be autowireable. so i think that should stay.

Comment (glaubinix) [reply src/DependencyInjection/HttplugExtension.php]: I think I misunderstood what this when I first adjusted the xml config. Both legacy XML files triggered a deprecation if `Http\Client\HttpClient` was used. I first read this the wrong way and thought it created a `httplug.client` alias.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ The change log describes what is "Added", "Removed", "Changed" or "Fixed" betwee
 - Changed the return type of `ClientFactory` to return a `ClientInterface` instead of `ClientInterface|HttpClient`
 - Changed the type of `httplug.client.default` to `ClientInterface` instead of `HttpClient`
 - Removed the `DummyClient` interface
+- Removed the `Http\Client\HttpClient` alias use the `ClientInterface` typehint in your services for autowiring.
```
Comment (dbu) [inline CHANGELOG.md]:
```suggestion
- Removed the `Http\Client\HttpClient` alias use the `Psr\Http\Client\ClientInterface` typehint in your services for autowiring.
```

**Review by dbu — APPROVED** [review]
thank you
